crates/transpiler/src/target/mod.rs

```diff
@@ -1410,19 +1410,32 @@ impl Target {
         let Some(qargs) = self.qargs() else {
             return Err(TargetCouplingError::AllToAll);
         };
+        let mut multi_q = false;
         for qargs in qargs {
             let Qargs::Concrete(qargs) = qargs else {
-                return Err(TargetCouplingError::AllToAll);
+                if self.global_operations.keys().any(|x| *x == 2) {
+                    return Err(TargetCouplingError::AllToAll);
+                }
+                if self.global_operations.keys().any(|x| *x > 2) {
+                    multi_q = true;
+                }
+                continue;
             };
```

Comment:
I'm not convinced that this method should silently permit a coupling map to be created in the case that there's a global explicitly multi-q operation, even if there are only concrete 2q operations - I feel like that behaviour is making quite an assumption and should be explicitly opt-in.  I get that the Python version ignored it, but I don't think the Python version necessarily made all the right choices.

Reply:
That's fair, it was a choice but is the documented behavior of the target's coupling map building. The original idea was that coupling map was the "legacy" view of the target which was incomplete so when you requested it you got the subset of data that you could represent in the legacy view. This is the same as `operations`/`basis_gates` which doesn't show the full picture. It actually used to emit a runtime warning that things were incomplete but that was deemed too noisy and we removed that.

I get that it may or may not have been the best choice, but I figured preserving the behavior was important for backwards compatibility, since with the current state of main if you had a target with both `cx` and `ccx` the behavior of the transpiler would change from 2.0 and 2.1 to 2.2.

Reply:
But obviously the only hard blocker I had in this PR was the 1q case which is incorrect. I only did the 3+ q case to mirror the behavior of the Python version before it because I had to change the logic in the code already.